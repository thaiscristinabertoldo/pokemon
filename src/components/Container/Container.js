import React, { memo } from 'react'

import ContainerStyled from './container.css'

export const Container = memo(({ children, ...props }) => {
  return <ContainerStyled {...props}>{children}</ContainerStyled>
})
