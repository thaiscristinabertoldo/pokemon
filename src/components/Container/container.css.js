import styled from 'styled-components'

const ContainerStyled = styled.div`
  display: flex;
  width: 100%;
  flex-wrap: wrap;
  flex-direction: ${(props) => props.flexDirection};
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  margin: ${(props) => props.margin};
`

ContainerStyled.defaultProps = {
  flexDirection: 'column',
  justifyContent: 'flex-start',
  alignItems: 'flex-start',
  margin: 0,
}

export default ContainerStyled
