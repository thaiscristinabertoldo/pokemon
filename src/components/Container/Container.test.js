import React from 'react'

import { render, cleanup } from '@testing-library/react'

import { Container } from './Container'

afterEach(cleanup)

describe('Container', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<Container />)

    expect(asFragment(<Container />)).toMatchSnapshot()
  })

  it('should take a children', () => {
    const { getByText } = render(
      <Container>
        <span>childrenText</span>
      </Container>
    )

    expect(getByText('childrenText')).toBeTruthy()
  })
})
