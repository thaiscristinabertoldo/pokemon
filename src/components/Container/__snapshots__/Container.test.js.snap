// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm hmTpfC"
  />
</DocumentFragment>
`;
